### PR TITLE
Enable Jenkins Security Scan

### PR DESCRIPTION
**Pull request by strangelookingnerd**
It is stated in the `JENKINS SECURITY POLICY` that

> The Jenkins project takes security seriously. We make every possible effort to ensure users can adequately secure their automation infrastructure. To that end, we work with Jenkins core and plugin developers, as well as security researchers, to fix security vulnerabilities in Jenkins in a timely manner, and to improve the security of Jenkins in general.

We as plugin developers should try our best to support this. One way of doing so is to enable the Jenkins Security Scan on this repository. For more details see the [documentation](https://www.jenkins.io/doc/developer/security/scan/).

This PR may create new issues in the Code scanning section of this project once it is merged. I'd be happy to try and help resolve those if necessary.

_Please note this PR was created semi-automatically. If you find any issue with it, don't hesitate to ping me._
_For more details see the [Jenkins Developers Google Group](https://groups.google.com/u/1/g/jenkinsci-dev/c/cxjQWnZ0S58)._

### Submitter checklist
- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

File: .github/workflows/jenkins-security-scan.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  security-events: write
+  contents: read
+  actions: read
+
+jobs:
+  security-scan:
+    uses: jenkins-infra/jenkins-security-scan/.github/workflows/jenkins-security-scan.yaml@v2
+    with:
+      java-cache: 'maven' # Optionally enable use of a build dependency cache. Specify 'maven' or 'gradle' as appropriate.
+      # java-version: 21 # Optionally specify what version of Java to set up for the build, or remove to use a recent default.
```
Comment (mig42) [inline .github/workflows/jenkins-security-scan.yml:21]: The plugin is currently compiled using `<java.level>8</java.level>` in POM, so perhaps we should set it here as well.

```suggestion
      java-version: 8 # Optionally specify what version of Java to set up for the build, or remove to use a recent default.
```

**Review by mig42 — CHANGES_REQUESTED** [review]
Thank you so much for taking the time to add the Jenkins security scan. Before merging the PR and running the code analysis, there's a minor point to discuss regarding Java versions.

File: .github/workflows/jenkins-security-scan.yml (hunk 1 above)
Comment (strangelookingnerd) [reply .github/workflows/jenkins-security-scan.yml:21]: > The plugin is currently compiled using `<java.level>8</java.level>` in POM, so perhaps we should set it here as well.

You are right, at the moment the plugin can not be compiled with Java 17 or 21.
While changing the version for the action would work, I'd propose to upgrade the plugins baseline alongside the required Java version. I'd be happy to provide a PR for that.

Comment (mig42) [reply .github/workflows/jenkins-security-scan.yml:21]: I see your point, I like your approach better. Now that #76 is merged, we can leave this as you submitted it. I'll solve this conversation.

**Review by mig42 — APPROVED** [review]
LGTM. Thanks @strangelookingnerd for these contributions!

**Pull request merged**